versions/3.1.1.md

```diff
@@ -2810,7 +2818,12 @@ components:
             type: boolean
 ```
 
-a payload like this:
+The `MyResponseType` schema will use the discriminator defined by the `Pet`
```

Comment:
This usage of `discriminator` deep inside inheritance tree seems quite complicated to implement and even reason about 🤔 Like, why should we use `MyResponseType.oneOf` and not `Cat.allOf`.

Shouldn't we require having `oneOf`/`anyOf`/`allOf` on the same level? We already enforce it in json schema btw, see https://github.com/OAI/OpenAPI-Specification/pull/3137.

Reply:
> Shouldn't we require having `oneOf`/`anyOf`/`allOf` on the same level? We already enforce it in json schema btw, see https://github.com/OAI/OpenAPI-Specification/pull/3137.

Good point! It seems that change was incorrect and should be backed out.

Reply:
Added a PR with the revert: https://github.com/OAI/OpenAPI-Specification/pull/3405